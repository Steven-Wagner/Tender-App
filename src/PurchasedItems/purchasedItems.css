.list-of-items {
    margin: 0px 15px;
}

.purchased-product {
    margin: 15px 0;
    border: 1px solid black;
    border-radius: 5px;
    box-shadow: 2px 2px 5px black;
    padding: 5px 5px;
}

.purchased-product:nth-child(even) {
    background-color: rgb(105, 188, 236);
}
.purchased-product:nth-child(odd) {
    background-color: rgb(108, 174, 212);
}

.purchased-info {
    display: flex;
    justify-content: space-between;
}