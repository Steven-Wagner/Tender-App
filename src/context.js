import React from 'react';

const TenderContext = React.createContext();

export const ProductsProvider = TenderContext.Provider
export const ProductsConsumer = TenderContext.Consumer

export default TenderContext;