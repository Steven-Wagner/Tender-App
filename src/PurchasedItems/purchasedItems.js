import React, {Component} from 'react';
import Nav from '../Components/Nav/nav';
import PurchasedItem from './PurchasedItem/purchasedItem';
import TenderContext from '../context';
import './purchasedItems.css';

class PurchasedItems extends Component {

    static contextType = TenderContext;

    componentDidMount() {
        window.scrollTo(0, 0);
    }

    render() {

        const items = this.context.purchasedItems.map((item, i) => {
            return(<PurchasedItem 
                item={item} 
                key={item.title}
                index={i}
                handleDelete={this.context.handlePurchasedItemDelete}/>)
        })

        return(
            <div className="nav-space">
                <Nav currentComponent='PurchasedItems'/>
                <div className="page-wrapper">
                    <header>
                        <h2 className="page-header">Purchased Products</h2>
                    </header>
                    <div className="list-of-items">
                        {items}
                    </div>
                </div>
            </div>
        )
    }
}

export default PurchasedItems;