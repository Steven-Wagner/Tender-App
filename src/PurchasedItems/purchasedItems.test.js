import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import PurchasedItems from './purchasedItems';
import Enzyme, { mount } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import {ProductsProvider} from '../context';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faShoppingCart } from '@fortawesome/free-solid-svg-icons';
import ReactDOM from 'react-dom';

library.add(faShoppingCart);

PurchasedItems.prototype.componentDidMount = () => {};

Enzyme.configure({ adapter: new Adapter() });

const context = {
    purchasedItems: [
        {title: 'Your Product Test Title', 
        img: 'An img', 
        desription: 'A Description',
        ad: 'Homepage ads',
        sold: '2', 
        profit: '4', 
        price: '2',
        id: 1,
        bonus: '1.00'},

        {title: 'Your Product Test Title 2', 
        img: 'An img 2', 
        desription: 'A Description 2',
        ad: 'Homepage ads 2',
        sold: '4', 
        profit: '12', 
        price: '3',
        id: 2,
        bonus: '1.00'},
    ]
};

function makePurchasedItemsWrapper(context) {
    return mount(
        <Router>
            <ProductsProvider value={context}>
                <PurchasedItems/>
            </ProductsProvider>
        </Router>)
}

describe('Purchased Items UI renders correctly', () => {
    it('renders content without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Router>
                <ProductsProvider value={context}>
                    <PurchasedItems/>
                </ProductsProvider>
            </Router> , div);
    });
      
    it('PurchasedItems renders correctly', () => {
        const tree = makePurchasedItemsWrapper(context);
        expect(tree).toMatchSnapshot();
    });

    it('Correct amount of items are rendered', () => {
        const wrapper = makePurchasedItemsWrapper(context);
        expect(wrapper.find('.purchased-product').length).toBe(2);
        wrapper.unmount();
    })
});