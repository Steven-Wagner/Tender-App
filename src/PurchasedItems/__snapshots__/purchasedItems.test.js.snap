// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Purchased Items UI renders correctly PurchasedItems renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PurchasedItems>
      <div
        className="nav-space"
      >
        <Nav
          currentComponent="PurchasedItems"
        />
        <main
          className="page-wrapper"
        >
          <header>
            <h1
              className="page-header"
            >
              Purchased Products
            </h1>
          </header>
          <div
            className="list-of-items"
          >
            <PurchasedItem
              index={0}
              item={
                Object {
                  "ad": "Homepage ads",
                  "bonus": "1.00",
                  "desription": "A Description",
                  "id": 1,
                  "img": "An img",
                  "price": "2",
                  "profit": "4",
                  "sold": "2",
                  "title": "Your Product Test Title",
                }
              }
              key="Your Product Test Title"
            >
              <section
                className="purchased-product"
              >
                <h2
                  className="product-title"
                >
                  Your Product Test Title
                </h2>
                <img
                  alt="Your Product Test Title"
                  className="product-pic"
                  src="An img"
                />
                <p
                  className="description"
                />
                <div
                  className="purchased-info"
                >
                  <p>
                    Price: 
                    2
                  </p>
                  <p>
                    Bonus: 
                    1.00
                  </p>
                </div>
              </section>
            </PurchasedItem>
            <PurchasedItem
              index={1}
              item={
                Object {
                  "ad": "Homepage ads 2",
                  "bonus": "1.00",
                  "desription": "A Description 2",
                  "id": 2,
                  "img": "An img 2",
                  "price": "3",
                  "profit": "12",
                  "sold": "4",
                  "title": "Your Product Test Title 2",
                }
              }
              key="Your Product Test Title 2"
            >
              <section
                className="purchased-product"
              >
                <h2
                  className="product-title"
                >
                  Your Product Test Title 2
                </h2>
                <img
                  alt="Your Product Test Title 2"
                  className="product-pic"
                  src="An img 2"
                />
                <p
                  className="description"
                />
                <div
                  className="purchased-info"
                >
                  <p>
                    Price: 
                    3
                  </p>
                  <p>
                    Bonus: 
                    1.00
                  </p>
                </div>
              </section>
            </PurchasedItem>
          </div>
        </main>
      </div>
    </PurchasedItems>
  </Router>
</BrowserRouter>
`;
